enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+#### Story 7
+
+As a Customer, I can use UWM without seeing Addons metrics, alerting or recording configurations.
+
+### Implementation Details/Notes/Constraints
+
+Since UWM and PM are part of Openshift Core, we would like to avoid increasing the complexity and introducing potential bugs for customers by extending those monitoring stacks. Because of this, we propose developing a new stack that will be able to support the managed service use-case while staying compatible with upcoming use-cases such as Hypershift or [KCP](https://github.com/kcp-dev/kcp).
+
+#### Overview
+
+We propose developing a new open-source operator with a working name, “MonitoringStack Operator” (official name proposals are welcome!) that manages a `MonitoringStack` CRD and can be used to deploy an independent, prometheus-based stack for each managed service.
+The operator will be installed into its own namespace and will watch `MonitoringStack` resources in the entire cluster. Alongside itself, the operator will deploy two other operators:
+
+* A Prometheus Operator in order to take advantage of the Prometheus management functionality already present in PO. An Alertmanager instance for managing alert routing will also be provisioned in the same namespace.
+* A [Grafana Operator](https://operatorhub.io/operator/grafana-operator)
+
+Once a `MonitoringStack` has been created, the operator will deploy the appropriate resources in the `MonitoringStackOperor`'s namespace based on how the stack is configured. In a simple case, this might be:
+
+* a Prometheus CR
+* a Grafana CR and a Grafana Data Source CR
+* a Thanos Querier if the stack is configured for high availability
+
+In other cases, it might be just a Prometheus Agent deployment (forwarding only capabilities).
+
+The `MonitoringStack` CRD is meant to be an abstraction over monitoring needs and requirements for a single tenant that specifies their needs. The underlying implementation is up to the MonitoringStackOperator.
+
+In this manner, managed service owners can simply create an instance of `MonitoringStack` alongside their services and delegate the provisioning and management of their monitoring stack to the operator.
+
+An example `MonitoringStack` resource could look as follows:
+
+```yaml
+---
+apiVersion: monitoring.coreos.io/v1alpha1
+kind: MonitoringStack
+metadata:
+    name: monitoring
+    namespace: redhat-kafka
+spec:
+    retention: 12d
+    resources:
+      requests: 
+        memory: 512M
+      limits:  
+        memory: 1G
+---
+```
+
+In the first iteration, each monitoring stack will be deployed and scoped to a single namespace. This means that the Prometheus instance will only be able to scrape monitoring resources from the namespace it is deployed in. In subsequent iterations, the CRD can be extended to allow scraping from multiple namespaces, either by namespace name or by namespace label selector.
```

Comment:
Writing the above also made me realise that we our current alerting depends on federation of metrics from cluster-monitoring, especially using kube-state-metrics. How do we address that need?

Reply:
Ah, I see this need was addressed under `Scraping metrics from platform components`